sklearn/ensemble/_gb.py

```diff
@@ -898,16 +898,16 @@ class GradientBoostingClassifier(ClassifierMixin, BaseGradientBoosting):
         Pass an int for reproducible output across multiple function calls.
         See :term:`Glossary <random_state>`.
 
-    max_features : int, float, string or None, optional (default=None)
+    max_features : {'auto', 'sqrt', 'log2'}, int or float, default=None
         The number of features to consider when looking for the best split:
 
         - If int, then consider `max_features` features at each split.
         - If float, then `max_features` is a fraction and
           `int(max_features * n_features)` features are considered at each
           split.
-        - If "auto", then `max_features=sqrt(n_features)`.
-        - If "sqrt", then `max_features=sqrt(n_features)`.
-        - If "log2", then `max_features=log2(n_features)`.
+        - If 'auto', then `max_features=sqrt(n_features)`.
+        - If 'sqrt', then `max_features=sqrt(n_features)`.
+        - If 'log2', then `max_features=log2(n_features)`.
```

Comment:
Are these changes necessary?

Reply:
Yes, perhaps I am overzealous here.

The story of this change was so:

First, I changed the `max_features` type from `str` to the set of allowed parameters. The contributing guide  uses single ticks, so the default  set was ``{'auto', 'sqrt', 'log2'}``.

Second, I changed the parameter description for consistency to single ticks. 

Shall I revert this? Just let me know.

Reply:
> The contributing guide uses single ticks, so the default set was {'auto', 'sqrt', 'log2'}.

I'm not sure there is a reason for it though. For instance, black will format with double tick in code. We can keep it as is here, but best to avoid such changes in future PRs.

Reply:
@rth I will, thanks for pointing this out!

Does sciki-learn use `black` for formatting code (even optionally)?

I am a great fan of the tool, so if this was the case I would definitely `blackify` my further contribution (if any).